smithy-diff/src/main/java/software/amazon/smithy/diff/evaluators/ChangedShapeType.java

```diff
@@ -33,12 +34,23 @@ public List<ValidationEvent> evaluate(Differences differences) {
         return differences.changedShapes()
                 .filter(diff -> diff.getOldShape().getType() != diff.getNewShape().getType())
                 .filter(diff -> !expectedSetToListChange(diff))
+                .filter(diff -> !expectedStringToEnumChange(diff))
                 .map(diff -> error(diff.getNewShape(), String.format(
                         "Shape `%s` type was changed from `%s` to `%s`.",
                         diff.getShapeId(), diff.getOldShape().getType(), diff.getNewShape().getType())))
                 .collect(Collectors.toList());
     }
 
+    static boolean expectedStringToEnumChange(ChangedShape<Shape> diff) {
+        ShapeType oldType = diff.getOldShape().getType();
+        ShapeType newType = diff.getNewShape().getType();
+
+        // Smithy diff doesn't raise an issue if a string with an enum trait is changed
+        // to an enum shape. The enum trait is deprecated and this is a recommended change.
+        return oldType == ShapeType.STRING && diff.getOldShape().hasTrait(EnumTrait.class)
+                && newType == ShapeType.ENUM;
```

Comment:
It might not make much of a difference but enum traits can only be used with strings, thus the test `oldType == ShapeType.STRING` is redundant, also given that the method is similar to the one below the logic can be written using the same style for consistency, i.e.,

```
    if (oldType == ShapeType.STRING && newType == ShapeType.ENUM) {
        return diff.getOldShape().hasTrait(EnumTrait.class);
    }
    return false;
```

Reply:
The current implementation matches closer what we are checking for semantically:

- `StringShape` with `@enum` trait in the old model
- `EnumShape` in the new model